### PR TITLE
Add utility required for Mongo-backed-storage tests.

### PR DESCRIPTION
**Pull request by poodlewars**
We use Mongo for some of our oldest libraries internally and need to test it still works.

I split the pytest build steps by Windows/Linux since the `mongodb` service container can only run on Linux and the Github Actions schema does not allow a "conditional" service container. The steps had actually already diverged quite a lot and the duplication is actually quite minimal. I pulled out the only chunky bit of logic to a composite action.

An alternative would have been to install `mongod` in the runners each time, but having tried that `pytest-server-fixtures` still could not launch the `mongod` instance correctly. Debugging the cause would have been extremely time consuming, and we prefer to test within an unmodified manylinux image anyway, for greater fidelity with user's environments.

To note, we set `CI_MONGO_HOST` to `mongodb` as that is the name of the service container, and GHA is putting the service container on _the same_ Docker network as the test container.

### CODE REVIEW

**Review by qc00 — APPROVED** [review]

File: build_tooling/build_many_linux_image.sh
Comment (poodlewars) [reply build_tooling/build_many_linux_image.sh:49]: Hmm it's for running the Python tests

File: build_tooling/mongodb.repo
Comment (qc00) [inline build_tooling/mongodb.repo:1]: Let's hope mongo's mirrors don't block us as there will be ~30 installs per build

File: build_tooling/mongod.sh
Comment (qc00) [inline build_tooling/mongod.sh:3]: Build reported

    cp: cannot create regular file ‘/etc/yum/repos.d/’: Not a directory

File: build_tooling/mongodb.repo
Comment (poodlewars) [reply build_tooling/mongodb.repo:1]: Yeah I'm going to fiddle around with this this afternoon, maybe adding a github package.

File: .github/workflows/build_steps.yml
Comment (qc00) [inline .github/workflows/build_steps.yml]: I think it's more trouble than it's worth to split this job by OS because we will soon have a Mac build, and the copies will start to diverge in dangerous ways.

Comment (poodlewars) [reply .github/workflows/build_steps.yml]: There's no choice, the containerized service can only run on Linux

File: .github/actions/install-wheel/action.yml
Comment (poodlewars) [reply .github/actions/install-wheel/action.yml:9]: This is all work in progress, there's no need to comment on it as it goes along

File: .github/workflows/build_steps.yml
Comment (qc00) [reply .github/workflows/build_steps.yml]: Ah, that's trivial to work around. Because the `services` key interprets expressions, you can do something like

```
services: ${{fromJSON(matrix.test_services)}}
```

Where `test_services` can be `null` for Windows and `{"mongo": {"image": "...."}}` for Linux.

You have to double JSON-encoded the values and put them in those giant JSON blocks in `build.yml`

Comment (qc00) [reply .github/workflows/build_steps.yml]: I think it's rather confusing that each version's job group now has a weird skipped job in it:
image

Also, having minor differences in duplicated copies of code in an esoteric language is highly likely going to lead to maintenance issues and bugs. E.g. future maintainers will have to guess whether an inconsistency is intentional or due to carelessness. This is especially risky with tests as we can fool ourselves into believing something is being tested.

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (mehertz) [inline python/tests/integration/arcticdb/version_store/test_basic_version_store.py:46]: Very minor: We don't really use the `smoke test` terminology anywhere in the code AFAIK. I'm probably just call this `TEST_VERSION_STORES`.

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (mehertz) [inline python/tests/integration/arcticdb/version_store/test_basic_version_store.py:43]: Can we not test if the `CI_MONGO_HOST` env var is set? If we don't, what happens if you run this locally on Linux?

File: .github/workflows/build_steps.yml
Comment (mehertz) [reply .github/workflows/build_steps.yml]: Can I be a second opinion?

If they were otherwise identical I think the JSON hack is a great idea, but it looks like there's already a bunch of if statements? Like:

```
- name: Select Python (Linux)
```

```
- name: Windows Pagefile
```

```
- name: Collect crash dumps (Windows)
```

At this point it's just a question of 1 big if or many smaller ifs and I think it's probably much of a muchness as to which is "better".

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (poodlewars) [reply python/tests/integration/arcticdb/version_store/test_basic_version_store.py:43]: No we shouldn't, since we don't insist on that being set for local runs (which work fine using pytest-server-fixtures as long as `mongod` is installed).

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (poodlewars) [reply python/tests/integration/arcticdb/version_store/test_basic_version_store.py:46]: I prefer it as is since _every_ version store in this file is really a "test version store".

**Review by willdealtry — APPROVED** [review]

File: .github/workflows/build.yml
Comment (qc00) [inline .github/workflows/build.yml:55]: Ah, this must have been confusing. I use it to divert my fork to run tests on my home cluster, so it's possible to debug the test job.

Let's remove it for now. I'll properly introduce it if we host our own runners.

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (qc00) [inline python/tests/integration/arcticdb/version_store/test_basic_version_store.py:44]: Curious. I thought the Windows runner already had Mongo service installed. You just need to add a command to start it

**Review by qc00 — APPROVED** [review]

**Pull request merged**
